apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibility.jsx

```diff
@@ -30,27 +30,60 @@ export default class AmazonFutureEngineerEligibility extends React.Component {
   constructor(props) {
     super(props);
 
-    let sessionEligibilityData =
-      JSON.parse(sessionStorage.getItem(sessionStorageKey)) || {};
+    let sessionEligibilityData = this.getSessionEligibilityData();
 
+    // Initial state is set by information that has been stored to the session.
+    // If none exists (ie, a user's first time visiting the page),
+    // set defaults that will ask them to provide school information.
     this.state = {
       formData: {
+        ...sessionEligibilityData,
         signedIn: this.props.signedIn,
-        schoolEligible:
-          sessionEligibilityData.schoolEligible ||
-          this.props.schoolEligible ||
-          null,
+        schoolEligible: this.checkInitialSchoolEligibility(
+          sessionEligibilityData
+        ),
         schoolId:
           sessionEligibilityData.schoolId || this.props.schoolId || null,
-        consentAFE: sessionEligibilityData.consentAFE || false,
-        submitted: sessionEligibilityData.submitted || false
+        consentAFE: sessionEligibilityData.consentAFE || false
       },
       errors: {}
     };
   }
 
-  updateFormData = change => {
-    this.setState({formData: {...this.state.formData, ...change}});
+  // If a user has gone through the eligibility flow,
+  // or has school information associated with their account,
+  // we use that for determining their eligibilty.
+  // If the school information associated with their account
+  // is ineligible, we still allow them to provide their school information
+  // in case their account information is out of date.
+  // Redirect to ineligible page ('/afe/benefits') if ineligible.
+  // Otherwise, we ask them for their school information.
```

Comment:
a little confused on the flow here--so if they are ineligible will we ask them to re-fill out school info before this point?

Reply:
Yeah this is kind of weird, I can describe what I'm trying to do and you can see if you think this code actually does that. Idea is that if you have an eligible school associated with your account (which would come through props), then we forward you to the full form (like what we want to do with people coming through the CTA that you worked on). Chatted with Bryan, and he preferred that if a teacher's account school info was ineligible, we should still give them the option to input their school in case it's out of date or something.

If you've visited the page and already provided school info, then we use that (from session storage), regardless of whether you are eligible or not.

Reply:
gotcha. It looks here like if they are ineligible we will redirect them to benefits, so do we allow them to input their school before this method is called?

Reply:
You'd get redirected if you had visited the page before and provided your school information (and that school was an ineligible school). We save that school information to session storage when you submit it.

This is basically so you can't just reload the page and input a new school (although there's a pretty gaping hole in that you could just open a new tab, but I don't think this is supposed to be a foolproof form).